app/javascript/vue/App.vue

```diff
@@ -1,20 +1,26 @@
 <template>
   <div class="shop">
-    <book-shelf></book-shelf>
+    <book-shelf :books="books"/>
   </div>
 </template>
 
 <script>
-  import bookData from './books-data'
+  import booksData from './books-data'
   import BookShelf from './BookShelf'
 
   export default {
-    data: () => ({}),
+    data() {
+      return {
+        books: []
+      }
+    },
     components: {
       BookShelf
     },
     created() {
-      this.books = bookData.books
+      booksData.then((books) => {
+        this.books = books
+      })
```

Comment:
What will you do if there will be an error while fetching data through ajax? You should let your users know about that.

Reply:
👍

Reply:
Good idea @resivalex!